lms/djangoapps/courseware/access_utils.py

```diff
@@ -92,6 +135,12 @@ def check_start_date(user, days_early_for_beta, start, course_key, display_error
         if should_grant_access:
             return ACCESS_GRANTED
 
+        # Before returning a StartDateError, determine if the learner should be redirected to the enterprise learner
+        # portal by returning StartDateEnterpriseLearnerError instead.
+        request = get_current_request()
+        if request and enterprise_learner_enrolled(request, user, course_key):
```

Comment:
These lines were added since the last try.  The idea is that we should check that there even is a request in the current context, and fallback to the old behavior if not.  This was the full backtrace from the first deploy which indicated that this function was being called from a celery task which would have resulted in `get_current_request()` returning None:

```
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/celery/app/trace.py", line 477, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/newrelic/hooks/application_celery.py", line 99, in wrapper
    return wrapped(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/celery/app/trace.py", line 760, in __protected_call__
    return self.run(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/edx_django_utils/monitoring/internal/code_owner/utils.py", line 193, in new_function
    return wrapped_function(*args, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/tasks.py", line 187, in recalculate_subsection_grade_v3
    _recalculate_subsection_grade(self, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/tasks.py", line 256, in _recalculate_subsection_grade
    raise self.retry(kwargs=kwargs, exc=exc)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/celery/app/task.py", line 736, in retry
    raise_with_context(exc)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/tasks.py", line 241, in _recalculate_subsection_grade
    _update_subsection_grades(
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/tasks.py", line 318, in _update_subsection_grades
    course_structure = get_course_blocks(student, store.make_course_usage_key(course_key))
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/course_blocks/api.py", line 107, in get_course_blocks
    return get_block_structure_manager(starting_block_usage_key.course_key).get_transformed(
  File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/content/block_structure/manager.py", line 77, in get_transformed
    transformers.transform(block_structure)
  File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/content/block_structure/transformers.py", line 113, in transform
    self._transform_with_filters(block_structure)
  File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/content/block_structure/transformers.py", line 132, in _transform_with_filters
    block_structure.filter_topological_traversal(combined_filters)
  File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/content/block_structure/block_structure.py", line 729, in filter_topological_traversal
    for _ in self.topological_traversal(filter_func=filter_func, **kwargs):
  File "/edx/app/edxapp/edx-platform/openedx/core/lib/graph_traversals.py", line 329, in _traverse_generic
    should_yield_node = filter_func(current_node)
  File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/content/block_structure/transformer.py", line 228, in <lambda>
    return lambda block_key: accumulated(block_key) and additional(block_key)
  File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/content/block_structure/transformer.py", line 228, in <lambda>
    return lambda block_key: accumulated(block_key) and additional(block_key)
  File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/content/block_structure/block_structure.py", line 685, in retain_or_remove
    if removal_condition(block_key):
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/course_blocks/transformers/start_date.py", line 100, in _removal_condition
    return not check_start_date(
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/access_utils.py", line 102, in check_start_date
    enterprise_customer_data = enterprise_customer_from_session_or_learner_data(get_current_request())
  File "/edx/app/edxapp/edx-platform/openedx/features/enterprise_support/api.py", line 385, in wrapper
    return func(*args, **kwargs)
  File "/edx/app/edxapp/edx-platform/openedx/features/enterprise_support/api.py", line 859, in enterprise_customer_from_session_or_learner_data
    enterprise_customer = enterprise_customer_from_session(request)
  File "/edx/app/edxapp/edx-platform/openedx/features/enterprise_support/api.py", line 426, in enterprise_customer_from_session
    return request.session.get(ENTERPRISE_CUSTOMER_KEY_NAME, _CACHE_MISS)
AttributeError: 'NoneType' object has no attribute 'session'
```